packages/jss/src/Jss.js

```diff
@@ -95,24 +99,28 @@ export default class Jss {
   /**
    * Create a rule without a Style Sheet.
    */
-  createRule(name?: string, style?: JssStyle = {}, options?: RuleFactoryOptions = {}): Rule {
+  createRule(name?: string, style?: JssStyle = {}, options?: RuleFactoryOptions = {}): Rule | null {
     // Enable rule without name for inline styles.
     if (typeof name === 'object') {
       return this.createRule(undefined, name, style)
     }
 
-    // Cast from RuleFactoryOptions to RuleOptions
-    // https://stackoverflow.com/questions/41328728/force-casting-in-flow
     const ruleOptions: RuleOptions = {...options, jss: this, Renderer: this.options.Renderer}
 
     if (!ruleOptions.generateClassName) ruleOptions.generateClassName = this.generateClassName
     if (!ruleOptions.classes) ruleOptions.classes = {}
     const rule = createRule(name, style, ruleOptions)
 
+    if (!rule) return null
+
     if (!ruleOptions.selector && rule instanceof StyleRule) {
       rule.selector = `.${ruleOptions.generateClassName(rule)}`
     }
 
+    if (rule instanceof KeyframesRule) {
+      rule.id = ruleOptions.generateClassName(rule)
```

Comment:
This doesn't feel in the right place, same with setting the selector. Can we maybe move this to the dedicated rules?

Reply:
You mean move this logic to the rule itself?

Reply:
yeah, I think this would be the logical place

Reply:
its been there some time ago, was not nice too

Reply:
The way I think about rules is that they have no knowledge of generated selectors, they only get them from the outside, no matter generated or not

Reply:
I think rules should generate everything that they need to work. In the case of StyleRule and KeyframesRule now that would be a selector

Reply:
They don't really care if selector was generated or not. In cases where rules are inside of `@global` selector needs to be global, in other cases  selector needs to be generated.